### PR TITLE
Battle ship

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
Port of battleship transaction family from 0.7 to 0.8.

### CODE REVIEW

**Comment by aludvik** [comment]
Need to clean up git commits and add tests

File: bin/run_lint [hunk 1]
```diff
@@ -160,12 +160,16 @@ lint rest_api/sawtooth_rest_api "$SINCE" || retval=1
...
 PYTHONPATH=$top_dir/families/identity
+PYTHONPATH=$top_dir/families/block_info
+PYTHONPATH=$top_dir/families/battleship
 PYTHONPATH=$PYTHONPATH:$top_dir/signing
 PYTHONPATH=$PYTHONPATH:$top_dir/families/supplychain/python
 PYTHONPATH=$PYTHONPATH:$top_dir/sdk/python
 export PYTHONPATH
 lint families/settings "$SINCE" || retval=1
 lint families/identity "$SINCE" || retval=1
+lint families/block_info "$SINCE" || retval=1
```
Comment (aludvik) [inline bin/run_lint]: Don't lint block_info here, it is being linted below.

File: bin/run_lint [hunk 2]
```diff
@@ -160,12 +160,16 @@ lint rest_api/sawtooth_rest_api "$SINCE" || retval=1
...
+lint families/battleship "$SINCE" || retval=1
```
Comment (aludvik) [inline bin/run_lint]: Copy the pattern for block_info below, don't add this to the list of families.

File: docker/compose/sawtooth-default.yaml [hunk 3]
```diff
@@ -36,6 +36,8 @@ services:
     entrypoint: intkey-tp-python -vv tcp://validator:4004
 
   xo-tp-python:
+    volumes:
+      - ../../:/project/sawtooth-core
```
Comment (aludvik) [inline docker/compose/sawtooth-default.yaml]: Leave this file alone.

File: bin/build_all [hunk 4]
```diff
@@ -253,6 +253,7 @@ then
         docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-settings-tp
         docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-identity-tp
         docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-block-info-tp
+        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-battleship-tp
```
Comment (aludvik) [inline bin/build_all]: Add a line to the other `build_python()` below, just copy the block_info and change.

File: families/battleship/sawtooth_battleship/processor/battleship_transaction.py [hunk 5]
```diff
@@ -229,7 +229,8 @@ def apply(self, state_store):
         Args:
             store: state_store
         """
-
+        LOGGER.debug('in apply')
+
```
Comment (aludvik) [inline families/battleship/sawtooth_battleship/processor/battleship_transaction.py]: This should be part of another commit.

File: families/battleship/sawtooth_battleship/processor/battleship_transaction.py [hunk 6]
```diff
@@ -227,12 +227,10 @@ def apply(self, state_store):
...
         Args:
-            store: state_store
-        """
-        LOGGER.debug('in apply')
-        
+            store: state_store"""
+
         state = _get_state_data(self._battleship_addr, state_store)
-        LOGGER.debug('Applying changes to state\nCURRENT STATE:\n%s' % state)
+        LOGGER.debug('Applying changes to state\nCURRENT STATE:\n%s', state)
```
Comment (aludvik) [inline families/battleship/sawtooth_battleship/processor/battleship_transaction.py]: Squash these type diffs into the original commit.

File: bin/build_all (hunk 4 above)
Comment (RyanLassigBanks) [reply bin/build_all]: Fixed

File: families/battleship/sawtooth_battleship/processor/battleship_transaction.py (hunk 5 above)
Comment (RyanLassigBanks) [reply families/battleship/sawtooth_battleship/processor/battleship_transaction.py]: Fixed

File: bin/run_lint (hunk 1 above)
Comment (RyanLassigBanks) [reply bin/run_lint]: Fix

File: bin/run_lint (hunk 2 above)
Comment (RyanLassigBanks) [reply bin/run_lint]: Fixed

File: docker/compose/sawtooth-default.yaml (hunk 3 above)
Comment (RyanLassigBanks) [reply docker/compose/sawtooth-default.yaml]: Fixed

File: bin/run_lint (hunk 1 above)
Comment (aludvik) [reply bin/run_lint]: Squash the change into this commit

File: docker/compose/sawtooth-default.yaml (hunk 3 above)
Comment (aludvik) [reply docker/compose/sawtooth-default.yaml]: Squash the change into this commit

File: bin/build_all (hunk 4 above)
Comment (aludvik) [reply bin/build_all]: Squash the change into this commit

File: families/battleship/sawtooth_battleship/__init__.py [hunk 7]
```diff
@@ -13,15 +13,11 @@
 # limitations under the License.
 # -----------------------------------------------------------------------------
 
-from sawtooth_battleship.txn_family import _register_transaction_types
+# from sawtooth_battleship.txn_family import _register_transaction_types
```
Comment (boydjohnson) [inline families/battleship/sawtooth_battleship/__init__.py]: Looks like this line can just be deleted.

File: bin/build_all (hunk 4 above)
Comment (RyanLassigBanks) [reply bin/build_all]: broken up and squashed
